packages/cloud-cognitive/src/components/TagSet/TagSet.js

```diff
@@ -18,189 +18,197 @@ import { pkg } from '../../settings';
-        } else {
-          break;
+          if (spaceAvailable >= tagWidth) {
+            spaceAvailable -= tagWidth;
+            willFit += 1;
+          } else {
+            break;
+          }
         }
-      }
 
-      if (willFit < sizingTags.length) {
-        while (
-          willFit > 0 &&
-          spaceAvailable < overflowTag.current.offsetWidth
-        ) {
-          // Highly unlikely any useful tag is smaller
-          willFit -= 1; // remove one tag
-          spaceAvailable += sizingTags[willFit].offsetWidth;
+        if (willFit < sizingTags.length) {
+          while (
+            willFit > 0 &&
+            spaceAvailable < overflowTag.current.offsetWidth
+          ) {
+            // Highly unlikely any useful tag is smaller
+            willFit -= 1; // remove one tag
+            spaceAvailable += sizingTags[willFit].offsetWidth;
+          }
         }
       }
-    }
 
-    if (willFit < 1) {
-      setDisplayCount(0);
-    } else {
-      setDisplayCount(maxVisible ? Math.min(willFit, maxVisible) : willFit);
-    }
-  }, [maxVisible, sizingTags]);
+      if (willFit < 1) {
+        setDisplayCount(0);
+      } else {
+        setDisplayCount(maxVisible ? Math.min(willFit, maxVisible) : willFit);
+      }
+    }, [maxVisible, sizingTags, tagSetRef]);
 
-  useEffect(() => {
-    checkFullyVisibleTags();
-  }, [checkFullyVisibleTags, maxVisible, sizingTags]);
+    useEffect(() => {
+      checkFullyVisibleTags();
+    }, [checkFullyVisibleTags, maxVisible, sizingTags]);
 
-  const handleResize = () => {
-    /* istanbul ignore next */ // not sure how to test resize
-    checkFullyVisibleTags();
-  };
+    const handleResize = () => {
+      /* istanbul ignore next */ // not sure how to test resize
+      checkFullyVisibleTags();
+    };
 
-  const handleSizerTagsResize = () => {
-    /* istanbul ignore next */ // not sure how to test resize
-    checkFullyVisibleTags();
-  };
+    const handleSizerTagsResize = () => {
+      /* istanbul ignore next */ // not sure how to test resize
+      checkFullyVisibleTags();
+    };
 
-  const handleModalClose = () => {
-    setShowAllModalOpen(false);
-  };
+    const handleModalClose = () => {
+      setShowAllModalOpen(false);
+    };
 
-  return (
-    <ReactResizeDetector onResize={handleResize}>
-      <div className={cx([blockClass, className])} ref={tagSet}>
-        <div
-          className={cx([
-            `${blockClass}__space`,
-            { [`${blockClass}__space--right`]: rightAlign },
-          ])}>
-          <ReactResizeDetector onResize={handleSizerTagsResize}>
-            <div
-              className={`${blockClass}__tag-container ${blockClass}__tag-container--hidden`}
-              aria-hidden={true}>
-              {hiddenSizingTags}
-            </div>
-          </ReactResizeDetector>
+    return (
+      <ReactResizeDetector onResize={handleResize}>
+        <div className={cx([blockClass, className])} ref={tagSetRef} {...rest}>
```

Comment:
Won't actually make difference here, but good practice to put the `{...rest}` first so that other props definitely override user passed props.